src/MultiTypeSymEntry.chpl

```diff
@@ -401,6 +402,36 @@ module MultiTypeSymEntry
         }
     }
 
+    class GroupBySymEntry:GenSymEntry {
+
+        var keyNamesEntry: shared SymEntry(string);
+        var keyTypesEntry: shared SymEntry(string);
+        var segmentsEntry: shared SymEntry(int);
+        var permEntry: shared SymEntry(int);
+        var ukIndEntry: shared SymEntry(int);
+        
+        proc init(keyNamesEntry: shared SymEntry, keyTypesEntry: shared SymEntry, segmentsSymEntry: shared SymEntry, 
+                    permSymEntry: shared SymEntry, ukIndSymEntry: shared SymEntry, itemsize: int) {
+            super.init(int, permSymEntry.size); // sets this.size = permEntry.size
+            this.entryType = SymbolEntryType.GroupBySymEntry;
+            assignableTypes.add(this.entryType);
+            this.keyNamesEntry = keyNamesEntry;
+            this.keyTypesEntry = keyTypesEntry;
+            this.segmentsEntry = segmentsSymEntry;
+            this.permEntry = permSymEntry;
+            this.ukIndEntry = ukIndSymEntry;
+
+            this.dtype = DType.UNDEF; // not required by groupby, but needs initialization
+            this.itemsize = itemsize; // itemsize will vary and can be accessed from specific keys
+
+            this.ndim = this.segmentsEntry.size; // used as the number of groups
+        }
+
+        override proc getSizeEstimate(): int {
+            return this.segmentsEntry.getSizeEstimate() + this.permEntry.getSizeEstimate();
+        }
```

Comment:
I think `getSizeEstimate` should include the other array components as well (`keyNamesEntry`, `keyTypesEntry`, `ukIndEntry`)

Reply:
Agreed. I copied this over and forgot to update it. I will add that now